nil pointer dereference reconciling InitConfiguration

/kind bug

**What steps did you take and what happened:**
Create a KubeadmConfig with only an InitConfiguration in its spec. It reconciles successfully the first time, but on subsequent runs it panics the controller:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x1e4bb70]

goroutine 26 [running]:
testing.tRunner.func1(0xc000486100)
	/usr/local/opt/go/libexec/src/testing/testing.go:874 +0x3a3
panic(0x1fa51c0, 0x2df9230)
	/usr/local/opt/go/libexec/src/runtime/panic.go:679 +0x1b2
sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/controllers.(*KubeadmConfigReconciler).Reconcile(0xc0000a0fc0, 0x214738a, 0x7, 0x2159154, 0x16, 0x0, 0x0, 0x0, 0x0)
	/Users/spellegrino/Code/Go/src/sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/controllers/kubeadmconfig_controller.go:152 +0x550
sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/controllers.TestKubeadmConfigReconciler_Reconcile_GenerateCloudConfigData(0xc000486100)
	/Users/spellegrino/Code/Go/src/sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/controllers/kubeadmconfig_controller_test.go:405 +0x629
testing.tRunner(0xc000486100, 0x21e4ad8)
	/usr/local/opt/go/libexec/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/opt/go/libexec/src/testing/testing.go:960 +0x350
```

**What did you expect to happen:**
The second reconcile to be a no-op.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Cluster API version:
- Cluster API Bootstrap Provider Kubeadm version:
- Cluster API Infrastructure Provider version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

--- a/controllers/kubeadmconfig_controller.go
+++ b/controllers/kubeadmconfig_controller.go
@@ -148,7 +148,7 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		err = patchHelper.Patch(ctx, config)
 		return ctrl.Result{}, err
 	// If we've already embedded a time-limited join token into a config, but are still waiting for the token to be used, refresh it
-	case config.Status.Ready:
+	case config.Status.Ready && (config.Spec.JoinConfiguration != nil && config.Spec.JoinConfiguration.Discovery.BootstrapToken != nil):
 		token := config.Spec.JoinConfiguration.Discovery.BootstrapToken.Token
 
 		// gets the remote secret interface client for the current cluster

--- a/controllers/kubeadmconfig_controller_test.go
+++ b/controllers/kubeadmconfig_controller_test.go
@@ -398,7 +398,13 @@ func TestKubeadmConfigReconciler_Reconcile_GenerateCloudConfigData(t *testing.T)
 		t.Fatal("Expected status ready")
 	}
 	if cfg.Status.BootstrapData == nil {
-		t.Fatal("Expected status ready")
+		t.Fatal("Expected generated bootstrap data")
+	}
+
+	// Ensure that we don't fail trying to refresh any bootstrap tokens
+	_, err = k.Reconcile(request)
+	if err != nil {
+		t.Fatalf("Failed to reconcile:\n %+v", err)
 	}
 }
 

